interview-questions.md

```diff
@@ -8,50 +8,51 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: so a paramenter is the scope of your function essentially, and the argument is what you are using in your function to compare against? if thats a good way of putting it, thats how i remember it at least
 
-Researched answer:
+Researched answer: the best answer i could find to explain this, and helped me understand it better was that the parameters are the aliases for the values inside the function and the arguments are the actual values.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: so I am VERY FAMILIAR with this after spending abvout 5 hours on the third question of the code challenge questions........... 
+you need a function, im not sure if that is considered a parameter though... element is the first which is required, index is the second and is optional, and the last is array which is also optional.
 
-Researched answer:
+Researched answer: so I had all of them essentially correct but there is another optional thisValue that I assume it used with objects? im having trouble finding specific instances of this, so i assume it use scarcely used.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: from what i remember, .map iterates through an array and returns an array of the same length. and .filter only returns values that meet a specified condition, so not necessarily of the same length in return.
 
-Researched answer:
+Researched answer: i was spot on on this one, map returns an array that has the same length as the original and filter removes the elements that dont belong after the conditions are met
 
-1. What is iteration?
+4. What is iteration?
 
-Your answer:
+Your answer: in simple ways of thinking iteration is acting on a function over the values within the function. 
 
-Researched answer:
+Researched answer: performing an action however many times is needed in order to meet the condition specified within the function
```

Comment:
Iteration is when you want to do something to each element in a collection and doesn't necessarily need to meet a condition.